waves/project.py

```diff
@@ -1624,6 +1647,274 @@ def energy_losses(
+            potential = potential_energy_gwh * 1e3
+        if units == "gw":
+            losses = losses
+            potential = potential_energy_gwh
+
+        unit_map = {"kw": "kWh", "mw": "MWh", "gw": "GWh"}
+        name = f"Energy Losses ({'%' if not ratio else unit_map[units]})"
+        if by == "windfarm":
+            losses = losses.sum(axis=1).to_frame(name)
+            potential = potential.sum(axis=1).to_frame(name)
+
+        # Aggregate to the desired frequency level (nothing required for month-year)
+        if frequency == "annual":
+            losses = losses.reset_index(drop=False).groupby("year").sum().drop(columns=["month"])
+            potential = (
+                potential.reset_index(drop=False).groupby("year").sum().drop(columns=["month"])
+            )
+        elif frequency == "project":
+            if by == "turbine":
+                losses = losses.sum(axis=0).to_frame(name).T
+                potential = potential.sum(axis=0).to_frame(name).T
+            else:
+                losses = losses.values.sum()
+                potential = potential.values.sum()
+
+        if ratio:
+            return losses / potential
+
+        if aep:
+            if frequency != "project":
+                raise ValueError("`aep` can only be set to True, if `frequency`='project'.")
+            losses /= self.operations_years
+
+        if per_capacity is None:
+            return losses
+        return losses / self.capacity(per_capacity)
+
+    def technical_loss_ratio(self) -> float:
+        """Calculate technical losses based on the project type.
+
+        This method is adopted from ORCA where a 1% hysterisis loss is applied for fixed-bottom
+        turbines. For floating turbines, this is 1% hysterisis, 0.1% for onboard equipment, and 0.1%
+        for rotor misalignment (0.01197901 total).
+
+        Returns
+        -------
+        float
+            The technical loss ratio.
+        """
+        if self.turbine_type == "floating":
+            return 1 - (1 - 0.01) * (1 - 0.001) * (1 - 0.001)
+        else:
+            return 0.01
+
+    def electrical_loss_ratio(self) -> float:
+        """Calculate electrical losses based on ORBIT parameters.
+
+        Returns
+        -------
+        float
+            The electrical loss ratio.
+
+        """
+        depth = self.orbit_config_dict["site"]["depth"]
+        distance_to_landfall = self.orbit_config_dict["site"]["distance_to_landfall"]
+        # ORCA formula
+        electrical_loss_ratio = (
+            2.20224112
+            + 0.000604121 * depth
+            + 0.0407303367321603 * distance_to_landfall
+            + -0.0003712532582 * distance_to_landfall**2
+            + 0.0000016525338 * distance_to_landfall**3
+            + -0.000000003547544 * distance_to_landfall**4
+            + 0.0000000000029271 * distance_to_landfall**5
+        ) / 100
+
+        return electrical_loss_ratio
+
+    def total_loss_ratio(
```

Comment:
The docstring should have an extended note (new paragraph betwween the description sentence and parameters section) that outlines the calculation and references ORCA/related paper for the source of the methodology.